dartagnan/src/main/java/com/dat3m/dartagnan/wmm/axiom/ForceEncodeAxiom.java

```diff
@@ -0,0 +1,30 @@
+package com.dat3m.dartagnan.wmm.axiom;
+
+import com.dat3m.dartagnan.wmm.relation.Relation;
+import com.dat3m.dartagnan.wmm.utils.TupleSet;
+import org.sosy_lab.java_smt.api.BooleanFormula;
+import org.sosy_lab.java_smt.api.SolverContext;
+
+/*
+    This is a fake axiom that forces a relation to get encoded!
+ */
+public class ForceEncodeAxiom extends Axiom {
+    public ForceEncodeAxiom(Relation rel) {
+        super(rel);
+    }
+
+    @Override
+    public TupleSet getEncodeTupleSet(){
+        return rel.getMaxTupleSet();
+    }
+
+    @Override
+    public BooleanFormula consistent(SolverContext ctx) {
+        return ctx.getFormulaManager().getBooleanFormulaManager().makeTrue();
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Do we really need a `toString` method? The only use case I see for this is that if someone prints the WMM, it will see that the relation is being forceEncoded (i.e. cuted), but since we are already logging this information, I don't really see the point. I would either just return "" or filter printing ForceEncodeAxioms when printing the WMM

Reply:
I don't see a benefit of not having a `toString` implementation. We don't print the `Wmm` for the user anyway. If we ever want to, we can change this. Also notice that the fake axiom is only placed on the artificially created baseline wmm. The original wmm is completely unaffected by all computations in this PR.

Reply:
The default implementation of `toString` is pretty useless. It is only fine for classes that represent functions rather than data (e.g. our preprocessors, analyses, and visitors). That's why IDE's tend to have the capability to auto-generate `toString` methods that exhibit the inner class data (usually as a list of the form `field=value`).
And I do think that we override `toString` in 90% of cases. If not, we should definitely do so.